examples/core/Graphormer/train.py

```diff
@@ -0,0 +1,483 @@
+
+    def collate(self, samples):
+        graphs, labels = map(list, zip(*samples))
+        labels = th.stack(labels)
+
+        num_graphs = len(graphs)
+        num_nodes = [g.num_nodes() for g in graphs]
+        max_num_nodes = max(num_nodes)
+
+        attn_mask = th.zeros(num_graphs, max_num_nodes + 1, max_num_nodes + 1)
+        node_feat = []
+        in_degree, out_degree = [], []
+        path_data = []
+        dist = -th.ones(
+            (num_graphs, max_num_nodes, max_num_nodes), dtype=th.long
+        )
+
+        for i in range(num_graphs):
+            attn_mask[i, :, num_nodes[i] + 1 :] = 1
+
+            node_feat.append(graphs[i].ndata["feat"] + 1)
+
+            in_degree.append(
+                th.clamp(graphs[i].in_degrees() + 1, min=0, max=512)
+            )
+            out_degree.append(
+                th.clamp(graphs[i].out_degrees() + 1, min=0, max=512)
+            )
+
+            # path & spatial
+            path = graphs[i].ndata["path"]
+            path_len = path.size(dim=2)
+
+            # shape: [n, n, max_len]
+            max_len = 5
+            if path_len >= max_len:
+                shortest_path = path[:, :, 0:max_len]
+            else:
+                p1d = (0, max_len - path_len)
+                shortest_path = F.pad(path, p1d, "constant", -1)
+            p3d = (
+                0,
+                0,
+                0,
+                max_num_nodes - num_nodes[i],
+                0,
+                max_num_nodes - num_nodes[i],
+            )
+            shortest_path = F.pad(shortest_path, p3d, "constant", -1)
+            edata = graphs[i].edata["feat"] + 1
+            edata = th.cat(
+                (edata, th.zeros(1, edata.shape[1]).to(edata.device)), dim=0
+            )
+
+            path_data.append(edata[shortest_path])
+
+            dist[i, : num_nodes[i], : num_nodes[i]] = graphs[i].ndata["spd"]
+
+        # node feat padding
+        node_feat = pad_sequence(node_feat, batch_first=True)
+
+        # degree padding
+        in_degree = pad_sequence(in_degree, batch_first=True)
+        out_degree = pad_sequence(out_degree, batch_first=True)
+
+        return (
+            labels.reshape(num_graphs, -1),
+            attn_mask,
+            node_feat,
+            in_degree,
+            out_degree,
+            th.stack(path_data),
+            dist,
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--seed",
+        default=1,
+        type=int,
+        help="Please give a value for seed",
+    )
+    parser.add_argument(
+        "--batch_size",
+        default=16,
+        type=int,
+        help="Please give a value for batch_size",
+    )
+    args = parser.parse_args()
+
+    # multi-GPUs
+    accelerator = Accelerator()
+
+    # setting seeds
+    random.seed(args.seed)
+    th.manual_seed(args.seed)
+    if th.cuda.is_available():
+        th.cuda.manual_seed(args.seed)
```

Comment:
Do you need the following?

```python
np.random.seed(seed)
torch.cuda.manual_seed_all(seed)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
dgl.seed(seed)
```

Reply:
No, they have no influence on the performance.